includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -63,16 +85,66 @@ public static function get_some_detached_subscriptions() {
 						'customer_id'               => $subscription->get_meta( '_stripe_customer_id' ),
 						'change_payment_method_url' => $subscription->get_change_payment_method_url(),
 					];
-					if ( count( $detached_subscriptions ) >= 5 ) {
-						break;
-					}
 				}
 			}
 		}
 
 		// Cache the result for a day.
-		set_transient( self::DETACHED_SUBSCRIPTIONS_TRANSIENT_KEY, $detached_subscriptions, DAY_IN_SECONDS );
+		WC_Stripe_Database_Cache::set( self::DETACHED_SUBSCRIPTIONS_CACHE_KEY, $detached_subscriptions, DAY_IN_SECONDS );
 
 		return $detached_subscriptions;
 	}
+
+	/**
+	 * Builds a string containing messages about subscriptions that are detached from the customer.
+	 *
+	 * @param $subscriptions array An array of subscriptions that are detached from the customer.
+	 * @return string A string containing the messages to be displayed in the admin interface.
+	 */
+	public static function build_subscriptions_detached_messages( $subscriptions = [] ) {
+		$detached_messages = '';
```

Comment:
Non-blocking thought: is it worth adding an early return when `$subscriptions` is empty?
```suggestion
		$detached_messages = '';
		if ( empty( $subscriptions ) || ! is_array( $subscriptions ) ) {
			return '';
		}
```